Android/ZensoftHRCRM/app/src/main/java/com/erkprog/zensofthrcrm/ui/candidates/candidatesList/CandidatesPresenter.java

```diff
@@ -1,64 +1,58 @@
 package com.erkprog.zensofthrcrm.ui.candidates.candidatesList;
 
-import android.annotation.SuppressLint;
-import android.util.Log;
-
 import com.erkprog.zensofthrcrm.data.entity.Candidate;
 import com.erkprog.zensofthrcrm.data.entity.CandidatesResponse;
-import com.erkprog.zensofthrcrm.data.network.candidates.CandidatesRepository;
-import com.google.gson.GsonBuilder;
-
-import java.util.List;
+import com.erkprog.zensofthrcrm.data.network.test.RestServiceTest;
 
+import retrofit2.Call;
+import retrofit2.Callback;
 import retrofit2.Response;
 
-public class CandidatesPresenter implements CandidatesContract.Presenter, CandidatesRepository
-    .OnCandidatesLoadFinishedListener {
-
-  private static final String TAG = "mylog:CandidatesPresenter";
+public class CandidatesPresenter implements CandidatesContract.Presenter {
 
   private CandidatesContract.View mView;
-  private CandidatesRepository mRepository;
+  private RestServiceTest mApiService;
 
-
-  public CandidatesPresenter(CandidatesContract.View view, CandidatesRepository repository) {
+  public CandidatesPresenter(CandidatesContract.View view, RestServiceTest service) {
     mView = view;
-    mRepository = repository;
+    mApiService = service;
   }
 
   @Override
   public void loadCandidates() {
-    mRepository.getCandidatesFromJson(this);
+    mApiService.getCandidates().enqueue(new Callback<CandidatesResponse>() {
+      @Override
+      public void onResponse(Call<CandidatesResponse> call, Response<CandidatesResponse> response) {
+        if (isViewAttached() && response.isSuccessful() && response.body() != null) {
```

Comment:
Condition isViewAttached should be global for both successful response or not, please fix it. It may cause NullPointException